File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {

        // throw new UnsupportedOperationException("Delete this statement and write your own implementation.");
        int number = String.valueOf(numberToCheck).length();
        int dupe = numberToCheck, sum = 0;
        for (int i = 0; i <= number; i++) {
            sum = sum + (int)Math.pow((dupe % 10), number);
            dupe /= 10;
        }
        if (sum == numberToCheck) {
            return true;
        }
        return false;
    }

}

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"

bool is_armstrong_number(int candidate) {
    int sum = 0,dupe = candidate;
    int count = number_of_digits(candidate);
    while (dupe != 0) {
        int temp = dupe % 10;
        sum = sum + pow(temp,count);
        dupe = dupe / 10;
    }
    if (dupe == candidate) {
        return true;
    }
    return false;
}

int number_of_digits(int candidate) {
    int count;
    while (candidate != 0) {
        candidate = candidate / 10;
        count++;
    }
    return count;
}